tutorial/developing_packages/local_package_development_flow.rst

```diff
@@ -1,4 +1,302 @@
 .. _local_package_development_flow:
 
-Package development flow
-=========================
+Package Development Flow
+========================
+
+This section introduces the **Conan local development flow**, which allows you to work on
+packages in your local project directory without having to export the contents of the
+package to the Conan cache first.
+
+This local workflow encourages users to perform trial-and-error in a local sub-directory
+relative to their recipe, much like how developers typically test building their projects
+with other build tools. The strategy is to test the `conanfile.py` methods individually
+during this phase.
+
+Let's use this flow for the ``hello`` package we created in :ref:`the previous
+section<tutorial_creating_packages>`.
+
+Please clone the sources to recreate this project. You can find them in the `examples2.0
+repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/tutorial/developing_packages/local_package_development_flow
+
+You can check the contents of the folder:
+
+..  code-block:: text
+
+    .
+    ├── conanfile.py
+    └── test_package
+        ├── CMakeLists.txt
+        ├── conanfile.py
+        └── src
+            └── example.cpp
+
+conan source
+------------
+
+You will generally want to start with the :command:`conan source` command. The strategy
+here is that you’re testing your source method in isolation and downloading the files to a
+temporary sub-folder relative to the `conanfile.py`. This relative folder is defined by
+the `self.folders.source` property in the `layout()` method. In this case, as we are using
+the pre-defined `cmake_layout` we set the value with the `src_folder` argument.
+
+Let's have a look at the recipe's `source()` and `layout()` method:
+
+..  code-block:: python
+
+    ...
+
+    def source(self):
+        # Please be aware that using the head of the branch instead of an immutable tag
+        # or commit is not a good practice in general.
+        get(self, "https://github.com/conan-io/libhello/archive/refs/heads/main.zip", 
```

Comment:
then why not use a specific commit here in example?

Reply:
It's because it was simpler to use it through the tutorial... I think it's good enough for the moment